v3/helpers/swap_math.py

```diff
@@ -151,3 +151,16 @@ def get_next_sqrtPrice(ratioA, liq, amount, zeroForOne):
         sqrtPrice_next = get_next_price_amount1(ratioA, liq, amount, zeroForOne)
 
     return sqrtPrice_next
+
+def addPrice(
+        data: pl.DataFrame
+) -> pl.DataFrame:
+    """
+    Helper function to add a new column with sqrtPriceX96 converted to price
+    """
+    sqrtPrice_list = data.select(pl.col("sqrtPriceX96")).to_series().to_list()
+    price_list = [(int(i)/(2 ** 96))**2 for i in sqrtPrice_list]
```

Comment:
i think i would prefer floats here. sometimes prices are very small because they are inverted to your expectation/high supply.

i think you should also be able to apply map_batches